app/javascript/packs/check_all.js

```diff
@@ -0,0 +1,18 @@
+function checkAll() {
+    const cb = document.getElementById("building_id");
+    const boxes = document.querySelectorAll("#building_building_id_");
+    let i;
+
+    if (cb.checked) {
+        for (i = 0; i < boxes.length; i++) {
+            boxes[i].checked = true;
+        }
+    }
+}
+
+document.addEventListener("change", e => {
```

Comment:
we should probably refine the target here

Reply:
JavaScript was work in progress - it's done now and EventListener has been added to master checkbox.